src/Symfony/Component/Config/Builder/Generator.php

```diff
@@ -0,0 +1,348 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Builder;
+
+use Symfony\Component\Config\Definition\ArrayNode;
+use Symfony\Component\Config\Definition\BooleanNode;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\Config\Definition\EnumNode;
+use Symfony\Component\Config\Definition\FloatNode;
+use Symfony\Component\Config\Definition\IntegerNode;
+use Symfony\Component\Config\Definition\NodeInterface;
+use Symfony\Component\Config\Definition\PrototypedArrayNode;
+use Symfony\Component\Config\Definition\ScalarNode;
+use Symfony\Component\Config\Definition\VariableNode;
+
+/**
+ * Build classes to help generate valid config.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class Generator implements GeneratorInterface
+{
+    private $classes;
+
+    /**
+     * @return \Closure that will return the root config class
+     */
+    public function build(ConfigurationInterface $configuration, string $outputDir): \Closure
+    {
+        $this->classes = [];
+
+        $rootNode = $configuration->getConfigTreeBuilder()->buildTree();
+        $rootClass = new ClassBuilder('Config\\'.$rootNode->getPath(), $rootNode->getName());
+        $this->classes[] = $rootClass;
+
+        $this->buildNode($rootNode, $rootClass, $rootClass->getNamespace());
+
+        $this->writeClasses($outputDir);
+        $loader = \Closure::fromCallable(function () use ($outputDir, $rootClass) {
+            $str = $outputDir.\DIRECTORY_SEPARATOR.$rootClass->getFilePath().\DIRECTORY_SEPARATOR.$rootClass->getFilename();
+            require_once $str;
+
+            $className = $rootClass->getFqcn();
+
+            return new $className();
+        });
+
+        return $loader;
+    }
+
+    private function writeClasses(string $outputDir)
+    {
+        foreach ($this->classes as $class) {
+            $this->buildConstructor($class);
+            $this->buildToArray($class);
+            $dir = $outputDir.\DIRECTORY_SEPARATOR.$class->getFilePath();
+            @mkdir($dir, 0777, true);
+            file_put_contents($dir.\DIRECTORY_SEPARATOR.$class->getFilename(), $class->build());
+        }
+
+        $this->classes = [];
+    }
+
+    private function buildNode(NodeInterface $node, ClassBuilder $class, string $namespace)
+    {
+        if (!$node instanceof ArrayNode) {
+            throw new \LogicException('The node was expected to be an ArrayNode. This Configuration includes an edge case not supported yet.');
+        }
+
+        /** @var NodeInterface $child */
+        foreach ($node->getChildren() as $child) {
+            switch ($child) {
```

Comment:
```suggestion
            switch (true) {
```

Reply:
note, `$child` works at runtime, since `($child == true)` is `true`, but psalm is more strict about this and considers switch to use `===`

Reply:
also, `/** @var NodeInterface $child */` can be removed, as it's currently the only reason psalm is passing.

Reply:
`switch(true)` will work, But so did @wouterj's suggestion. Both are applied. 

Thank you

Reply:
`/** @var NodeInterface $child */` solution is actually a bug, and psalm shouldn't allow it, but it currently does.

reported in https://github.com/vimeo/psalm/issues/5494